JSQMessagesViewController/Model/JSQAudioMediaItem.m

```diff
@@ -0,0 +1,389 @@
+//
+//  JSQAudioMediaItem.m
+//
+
+#import "JSQAudioMediaItem.h"
+
+#import "JSQMessagesMediaPlaceholderView.h"
+#import "JSQMessagesMediaViewBubbleImageMasker.h"
+
+#import "UIImage+JSQMessages.h"
+#import "UIColor+JSQMessages.h"
+
+@interface JSQAudioMediaItem ()
+
+@property (strong, nonatomic) UIView *          cachedMediaView;
+
+@property (strong, nonatomic) UIButton *        playButton;
+
+@property (strong, nonatomic) UIProgressView *  progressView;
+@property (strong, nonatomic) UILabel *         progressLabel;
+@property (strong, nonatomic) NSTimer *         progressTimer;
+
+@property (strong, nonatomic) AVAudioPlayer *   audioPlayer;
+
+@end
+
+@implementation JSQAudioMediaItem
+
+#pragma mark - Initialization
+
+- (instancetype)initWithURL:(NSURL *)audioURL isReadyToPlay:(BOOL)isReadyToPlay
+{
+    self = [super init];
+    if (self) {
+        _audioURL = [audioURL copy];
+        _isReadyToPlay = isReadyToPlay;
+        _cachedMediaView = nil;
+        [self defaultsInit];
+    }
+    return self;
+}
+
+- (instancetype)initWithData:(NSData *)audioData
+{
+    self = [super init];
+    if (self) {
+        _audioData = [audioData copy];
+        _isReadyToPlay = [audioData length] > 0;
+        _cachedMediaView = nil;
+        [self defaultsInit];
+    }
+    return self;
+}
+
+- (void)defaultsInit
+{
+    // set defaults when initialized; user can override
+    
+    _controlPadding = 6;
+    _controlInsets = UIEdgeInsetsMake(6, 6, 6, 18);
+    
+    _labelFont = [UIFont systemFontOfSize:12];
+
+    _backgroundColor = [UIColor jsq_messageBubbleLightGrayColor];
+    _tintColor = [UIButton buttonWithType:UIButtonTypeSystem].tintColor;
+    
+    _audioCategory = @"AVAudioSessionCategoryPlayback";
+    _audioCategoryOptions = AVAudioSessionCategoryOptionDuckOthers | AVAudioSessionCategoryOptionDefaultToSpeaker | AVAudioSessionCategoryOptionAllowBluetooth;
+}
```

Comment:
We'll need to refine this initializer flow:

- we need a designated init, which should receive all params. this should be the `NSData` init
- the `NSURL` init can delegate to the `NSData` init (with `dataWithContentsOfURL:...`)
- implement `-init` with all defaults instead of `defaultsInit` (but `-init` should call the designated init)